BUILD.md

```diff
@@ -0,0 +1,69 @@
+# Building DLAF
+
+## Dependencies
+
+- MPI
+- [pika](https://github.com/pika-org/pika)
+- [umpire](https://github.com/LLNL/Umpire)
+- [blaspp](https://bitbucket.org/icl/blaspp/src/default/)
+- [lapackpp](https://bitbucket.org/icl/lapackpp/src/default/)
+- Intel MKL or other LAPACK implementation
+- A ScaLAPACK implementation (optional, ScaLAPACK-like C API only)
+- [whip](https://github.com/eth-cscs/whip) (optional, GPU only)
+- [cuda](https://developer.nvidia.com/cuda) (optional, NVidia GPUs only)
+- [HIP/ROCm](https://github.com/RadeonOpenCompute/ROCm) (optional, AMD GPUs only)
+- [GoogleTest](https://github.com/google/googletest) (optional; bundled) - unit testing
+- Doxygen (optional) - documentation
+
+## Build using Spack
+
+We provide a spack package `dla-future` that can be easily added to your own spack as follows:
+
+`spack repo add $DLAF_ROOT/spack`
+
+This will add a new repository with namespace `dlaf`.
```

Comment:
We can probably drop this since we have a public version of the spack package. IIRC for the end user it should not be relevant our internal version (even for master), right?

Reply:
#1138